### PR TITLE
[iOS] Fix almost all Xcode warnings

### PR DESCRIPTION
**Pull request by rtsisyk**

### CODE REVIEW

File: drape_frontend/poi_symbol_shape.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ void SolidBatch(ref_ptr<dp::GraphicsContext> context, ref_ptr<dp::Batcher> batch
   state.SetTextureFilter(dp::TextureFilter::Nearest);
   state.SetTextureIndex(symbolRegion.GetTextureIndex());
 
-  dp::AttributeProvider provider(1 /* streamCount */, vertexes.size());
+  dp::AttributeProvider provider(1 /* streamCount */, (uint32_t)vertexes.size());
```
Comment (vng) [inline drape_frontend/poi_symbol_shape.cpp:94]: I don't like this kind of suppressing. Better to leave as is or static_cast.

File: drape_frontend/poi_symbol_shape.cpp [hunk 2]
```diff
@@ -150,7 +150,7 @@ void MaskedBatch(ref_ptr<dp::GraphicsContext> context, ref_ptr<dp::Batcher> batc
...
```
Comment (vng) [inline drape_frontend/poi_symbol_shape.cpp:153]: I don't like this kind of suppressing. Better to leave as is or static_cast.

File: iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -895,8 +893,6 @@
 			isa = PBXResourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				479834F023426C0100724D1E /* common-debug.xcconfig in Resources */,
-				479834F123426C0100724D1E /* common-release.xcconfig in Resources */,
```
Comment (vng) [inline iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj:899]: Hm, really not needed here?

File: iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -997,6 +993,7 @@
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = NO;
```
Comment (vng) [inline iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj:996]: BTW, why don't we put all this CLANG_WARN_XXX into common-debug/common-release ?
Here and in every other pbxproj ..

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -6105,12 +6103,18 @@
 				ASSETCATALOG_COMPILER_LAUNCHIMAGE_NAME = LaunchImage;
 				BUNDLE_IDENTIFIER = app.omaps.debug;
 				CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = YES;
+				CLANG_UNDEFINED_BEHAVIOR_SANITIZER_NULLABILITY = NO;
```
Comment (vng) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6106]: Are you sure about this?

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -6119,21 +6123,22 @@
...
-					"$(inherited)",
-				);
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
+				LIBRARY_SEARCH_PATHS = "$(inherited)";
 				OTHER_CFLAGS = (
 					"$(inherited)",
 					"-Wall",
 					"-Wno-unused-local-typedef",
+					"-Wno-nonnull",
+					"-Wno-nullability-completeness",
```
Comment (vng) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6133]: Are you sure about this?

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -6119,21 +6123,22 @@
...
 					"-Wimplicit-retain-self",
 					"-fcxx-modules",
 				);
 				OTHER_CPLUSPLUSFLAGS = (
 					"$(OTHER_CFLAGS)",
 					"-fcxx-modules",
 				);
-				OTHER_SWIFT_FLAGS = "$(inherited)";
+				OTHER_SWIFT_FLAGS = "$(inherited) -Xcc -Wno-nullability-completeness";
```
Comment (vng) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6141]: Are you sure about this?

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -6143,7 +6148,9 @@
 				SWIFT_OBJC_INTERFACE_HEADER_NAME = SwiftBridge.h;
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
 				TARGETED_DEVICE_FAMILY = "1,2";
+				VALIDATE_WORKSPACE_SKIPPED_SDK_FRAMEWORKS = GLKit;
 				VERSIONING_SYSTEM = "apple-generic";
+				WARNING_CFLAGS = "";
```
Comment (vng) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6153]: ? Looks strange

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -6155,12 +6162,18 @@
 				ASSETCATALOG_COMPILER_LAUNCHIMAGE_NAME = LaunchImage;
 				BUNDLE_IDENTIFIER = app.omaps.release;
 				CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = YES;
+				CLANG_UNDEFINED_BEHAVIOR_SANITIZER_NULLABILITY = NO;
```
Comment (vng) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6165]: Are you sure about this?

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -6169,11 +6182,14 @@
...
 				);
 				INFOPLIST_FILE = OMaps.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				LIBRARY_SEARCH_PATHS = "$(inherited)";
 				OTHER_CFLAGS = (
 					"$(inherited)",
 					"-Wall",
 					"-Wno-unused-local-typedef",
+					"-Wno-nonnull",
+					"-Wno-nullability-completeness",
```
Comment (vng) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6192]: Are you sure about this?

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -6191,7 +6207,9 @@
 				SWIFT_OBJC_INTERFACE_HEADER_NAME = SwiftBridge.h;
 				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
 				TARGETED_DEVICE_FAMILY = "1,2";
+				VALIDATE_WORKSPACE_SKIPPED_SDK_FRAMEWORKS = GLKit;
 				VERSIONING_SYSTEM = "apple-generic";
+				WARNING_CFLAGS = "";
```
Comment (vng) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6212]: Looks strange

File: xcode/open_location_code/open_location_code.xcodeproj/project.pbxproj [hunk 12]
```diff
@@ -155,6 +156,32 @@
...
+				CLANG_WARN_UNREACHABLE_CODE = YES;
+				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
```
Comment (vng) [inline xcode/open_location_code/open_location_code.xcodeproj/project.pbxproj:184]: All this stuff can/should be in common-

**Review by vng — CHANGES_REQUESTED** [review]

File: iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj (hunk 3 above)
Comment (rtsisyk) [reply iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj:899]: I have no idea. I've changed the project by using Xcode UI. I will try to revert this change and see what happens.

File: iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj (hunk 4 above)
Comment (rtsisyk) [reply iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj:996]: I'm a newbie in Xcode project. Nice to learn that there is a common file which can override settings for all project. I will try to move all common settings to  common-debug/common-release. Thanks for the advice!

File: iphone/Maps/Maps.xcodeproj/project.pbxproj (hunk 5 above)
Comment (rtsisyk) [reply iphone/Maps/Maps.xcodeproj/project.pbxproj:6106]: No, I'm not certain.  My goal was to fix the most errors and re-force "zero-warning" policy. Unfortunately, I realized that clang's built sanitizer is too noisy. I agree that "nullability" checks are good in general.  Unfortunately, we don't have enough resources to resolve all problems of such kind. I think we need to preserve a status-quo and at least don't introduce new problems.

**Comment by rtsisyk** [comment]
I will rework this PR on the next week.

File: drape_frontend/poi_symbol_shape.cpp (hunk 1 above)
Comment (rtsisyk) [reply drape_frontend/poi_symbol_shape.cpp:94]: Reverted.

File: drape_frontend/poi_symbol_shape.cpp (hunk 2 above)
Comment (rtsisyk) [reply drape_frontend/poi_symbol_shape.cpp:153]: Revered.

File: iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj (hunk 3 above)
Comment (rtsisyk) [reply iphone/CoreApi/CoreApi.xcodeproj/project.pbxproj]: Reverted

File: iphone/Maps/Maps.xcodeproj/project.pbxproj (hunk 9 above)
Comment (rtsisyk) [reply iphone/Maps/Maps.xcodeproj/project.pbxproj:6164]: https://clang.llvm.org/docs/analyzer/developer-docs/nullability.html

Clang supports special `nonnull` attribute for pointers. Some systems libraries use this attribute. Our code is not ready for that.

File: xcode/open_location_code/open_location_code.xcodeproj/project.pbxproj (hunk 12 above)
Comment (rtsisyk) [reply xcode/open_location_code/open_location_code.xcodeproj/project.pbxproj]: I moved flags to common, but this part is probably autogenerated by Xcode.

**Comment by rtsisyk** [comment]
Re-worked. Please review.

File: iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm [hunk 13]
```diff
@@ -408,8 +408,9 @@ - (MWMBookmarkGroupType)getCategoryGroupType:(MWMMarkGroupID)groupId {
       return MWMBookmarkGroupTypeCollection;
     case kml::CompilationType::Day:
       return MWMBookmarkGroupTypeDay;
+    default:
+      return MWMBookmarkGroupTypeRoot;
   }
-  return MWMBookmarkGroupTypeRoot;
```
Comment (vng) [inline iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm:412]: It's not a good way to suppress this warning. If I will add new option like MWMBookmarkGroupTypeXXX, I will **not** get any warning here. If compiler gives warning here and we process **all** case variants here, we should do something else ..

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 14]
```diff
@@ -6105,12 +6103,17 @@
...
+				CLANG_WARN_COMMA = NO;
+				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/3party",
 				);
 				GCC_PREFIX_HEADER = Maps_Prefix.pch;
 				GCC_PREPROCESSOR_DEFINITIONS = "$(inherited)";
+				GCC_WARN_ABOUT_DEPRECATED_FUNCTIONS = NO;
+				GCC_WARN_CHECK_SWITCH_STATEMENTS = NO;
```
Comment (vng) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6116]: Controversial about switch.

File: iphone/Maps/Maps.xcodeproj/project.pbxproj (hunk 8 above)
Comment (vng) [reply iphone/Maps/Maps.xcodeproj/project.pbxproj:6152]: Let's remove WARNING_CFLAGS = ""; entry

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 15]
```diff
@@ -6155,12 +6161,17 @@
...
```
Comment (vng) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6174]: Controversial about switch.

File: iphone/Maps/Maps.xcodeproj/project.pbxproj (hunk 11 above)
Comment (vng) [reply iphone/Maps/Maps.xcodeproj/project.pbxproj:6210]: Let's remove WARNING_CFLAGS = ""; entry

File: partners_api/partners.cpp [hunk 16]
```diff
@@ -57,13 +57,16 @@ PartnerInfo::PartnerInfo(int partnerIndex, std::string && name)
...
+#pragma GCC diagnostic ignored "-Wunreachable-code"
 #if defined(OMIM_OS_IPHONE)
   return m_iosBannerPlacementId;
 #elif defined(OMIM_OS_ANDROID)
   return m_androidBannerPlacementId;
 #endif
   static std::string kEmptyStr;
   return kEmptyStr;
+#pragma GCC diagnostic pop
 }
```
Comment (vng) [inline partners_api/partners.cpp:70]: I would prefer to add
#define UNREACHABLE_CODE __builtin_unreachable

And use it here and below like UNREACHABLE_CODE();

File: shaders/metal_program_pool.mm [hunk 17]
```diff
@@ -112,9 +112,10 @@ MTLVertexFormat GetFormatByDataType(MTLDataType dataType)
   case MTLDataTypeFloat2: return MTLVertexFormatFloat2;
   case MTLDataTypeFloat3: return MTLVertexFormatFloat3;
   case MTLDataTypeFloat4: return MTLVertexFormatFloat4;
+  default:
+    CHECK(false, ("Unsupported vertex format."));
+    return MTLVertexFormatInvalid;
   }
-  CHECK(false, ("Unsupported vertex format."));
-  return MTLVertexFormatInvalid;
```
Comment (vng) [inline shaders/metal_program_pool.mm:117]: As I wrote before, switch warning should not be suppressed like this.

File: shaders/metal_program_pool.mm [hunk 18]
```diff
@@ -125,9 +126,10 @@ uint32_t GetSizeByDataType(MTLDataType dataType)
   case MTLDataTypeFloat2: return 2 * sizeof(float);
   case MTLDataTypeFloat3: return 3 * sizeof(float);
   case MTLDataTypeFloat4: return 4 * sizeof(float);
+  default:
+    CHECK(false, ("Unsupported vertex format."));
+    return 0;
   }
-  CHECK(false, ("Unsupported vertex format."));
-  return 0;
```
Comment (vng) [inline shaders/metal_program_pool.mm:130]: As I wrote before, switch warning should not be suppressed like this.

**Review by vng — CHANGES_REQUESTED** [review]

File: iphone/Chart/Chart.xcodeproj/project.pbxproj [hunk 19]
```diff
@@ -439,7 +439,7 @@
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
 				INFOPLIST_FILE = Chart/Info.plist;
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
-				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Comment (biodranik) [inline iphone/Chart/Chart.xcodeproj/project.pbxproj:442]: Зачем увеличивать минимально поддерживаемую версию iOS? Есть веские причины для этого?

File: iphone/Maps/Categories/UIColor+MapsMeColor.m [hunk 20]
```diff
@@ -6,10 +6,6 @@
 static NSDictionary<NSString *, UIColor *> *day;
 static NSDictionary<NSString *, UIColor *> *night;
 
-static UIColor * color(SEL cmd) {
```
Comment (biodranik) [inline iphone/Maps/Categories/UIColor+MapsMeColor.m:9]: Этот метод точно не используется?

File: iphone/Maps/Classes/MapsAppDelegate.mm [hunk 21]
```diff
@@ -672,7 +672,7 @@ - (void)onConversionDataReceived:(NSDictionary *)installData {
     NSURL *deeplinkUrl = [NSURL URLWithString:deeplink];
     if (deeplinkUrl != nil) {
       dispatch_async(dispatch_get_main_queue(), ^{
-        [[DeepLinkHandler shared] applicationDidReceiveUniversalLink:deeplinkUrl provider:DeepLinkProviderAppsflyer];
+        (void)[[DeepLinkHandler shared] applicationDidReceiveUniversalLink:deeplinkUrl provider:DeepLinkProviderAppsflyer];
```
Comment (biodranik) [inline iphone/Maps/Classes/MapsAppDelegate.mm:675]: Предлагаю выпилить AppsFlyer целиком с этим кодом.

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 22]
```diff
@@ -6089,10 +6088,9 @@
 				DEVELOPMENT_TEAM = W55CJ2SCF7;
 				"EXCLUDED_SOURCE_FILE_NAMES[sdk=iphonesimulator*]" = MetalContextFactory.mm;
 				"EXCLUDED_SOURCE_FILE_NAMES[sdk=macosx*]" = MetalContextFactory.mm;
-				FRAMEWORK_SEARCH_PATHS = (
-					"$(inherited)",
-				);
+				FRAMEWORK_SEARCH_PATHS = "$(inherited)";
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				MARKETING_VERSION = 10.5.1;
```
Comment (biodranik) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6093]: Почему под iOS сейчас именно такая версия?

File: iphone/Maps/Maps.xcodeproj/project.pbxproj (hunk 5 above)
Comment (biodranik) [reply iphone/Maps/Maps.xcodeproj/project.pbxproj:6106]: Do not do it if you are not certain. Let's revert it.

File: iphone/Maps/Maps.xcodeproj/project.pbxproj [hunk 23]
```diff
@@ -6105,12 +6103,17 @@
 				ASSETCATALOG_COMPILER_LAUNCHIMAGE_NAME = LaunchImage;
 				BUNDLE_IDENTIFIER = app.omaps.debug;
 				CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = YES;
+				CLANG_UNDEFINED_BEHAVIOR_SANITIZER_NULLABILITY = NO;
+				CLANG_WARN_COMMA = NO;
```
Comment (biodranik) [inline iphone/Maps/Maps.xcodeproj/project.pbxproj:6107]: Please also revert this line. Zero-warnings policy is done by fixing the source of warnings, not by silencing them.

File: iphone/Maps/Maps.xcodeproj/project.pbxproj (hunk 14 above)
Comment (biodranik) [reply iphone/Maps/Maps.xcodeproj/project.pbxproj:6116]: Please also revert these lines. Zero-warnings policy is done by fixing the source of warnings, not by silencing them.

File: iphone/Maps/Maps.xcodeproj/project.pbxproj (hunk 6 above)
Comment (biodranik) [reply iphone/Maps/Maps.xcodeproj/project.pbxproj:6132]: Please also revert these 2 lines. Zero-warnings policy is done by fixing the source of warnings, not by silencing them.

File: iphone/Maps/Maps.xcodeproj/project.pbxproj (hunk 7 above)
Comment (biodranik) [reply iphone/Maps/Maps.xcodeproj/project.pbxproj:6140]: Please revert.

File: kml/serdes.cpp [hunk 24]
```diff
@@ -594,7 +594,7 @@ bool ParsePoint(std::string const & s, char const * delim, m2::PointD & pt,
     {
       pt = mercator::FromLatLon(lat, lon);
 
-      double rawAltitude;
+      double rawAltitude = 0.0;
```
Comment (biodranik) [inline kml/serdes.cpp:597]: Правильнее было бы заглушить ворнинг в этой строке. Странно, что он вообще возник, ведь to_double рассчитан на неинициализированную переменную. Возможно, есть какие-то специфические прагмы?

File: partners_api/partners.cpp (hunk 16 above)
Comment (biodranik) [reply partners_api/partners.cpp:70]: Let's remove the unreachable code everywhere. It's not needed.

File: xcode/alohalitics/alohalitics.xcodeproj/project.pbxproj [hunk 25]
```diff
@@ -185,6 +188,7 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 34EBB4791DBF5159005BE9B8 /* common-release.xcconfig */;
 			buildSettings = {
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
```
Comment (biodranik) [inline xcode/alohalitics/alohalitics.xcodeproj/project.pbxproj:191]: Зачем в каждой конфигурации это прописывать? Предлагаю вообще не трогать файлы проектов, с ними нужно аккуратно, я потом сам поправлю.

File: xcode/mapshot/mapshot.xcodeproj/project.pbxproj [hunk 26]
```diff
@@ -362,12 +366,14 @@
...
 					"$(OMIM_ROOT)/3party/gflags/src/",
 				);
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 			};
 			name = "Production Full";
 		};
 		A8E5415D1F9FBDB300A1B8FA /* Production Full */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CODE_SIGN_IDENTITY = "-";
```
Comment (biodranik) [inline xcode/mapshot/mapshot.xcodeproj/project.pbxproj:376]: Не думаю, что это хорошая идея 1) выставить - и 2) прописать это явно в каждой конфигурации.

**Comment by rtsisyk** [comment]
@biodranik will make a proper patch to actually fix warnings instead of suppressing them.

**Pull request closed without merging**